sample-notebooks/All-hands Demo.ipynb

```diff
@@ -0,0 +1,494 @@
+    "    \"lambda_bias\": \"0.0\",\n",
+    "    \"max_bin\": \"256\",\n",
+    "    \"normalize_type\": \"tree\",\n",
+    "    \"objective\": \"reg:linear\",\n",
+    "    \"one_drop\": \"0\",\n",
+    "    \"prob_buffer_row\": \"1.0\",\n",
+    "    \"process_type\": \"default\",\n",
+    "    \"refresh_leaf\": \"1\",\n",
+    "    \"sample_type\": \"uniform\",\n",
+    "    \"scale_pos_weight\": \"1.0\",\n",
+    "    \"silent\": \"0\",\n",
+    "    \"skip_drop\": \"0.0\",\n",
+    "    \"tree_method\": \"auto\",\n",
+    "    \"tweedie_variance_power\": \"1.5\",\n",
+    "    \"updater\": \"grow_colmaker,prune\",\n",
+    "}\n",
+    "\n",
+    "# Data transformation task\n",
+    "transform_parquet_to_csv = SdkTask.fetch(project=\"flytesnacks\", domain=\"development\", name=\"transform_parquet_to_csv\", version=\"24\")\n",
+    "\n",
+    "environ[\"version\"] = \"training_39\"\n",
+    "environ[\"spec_version\"] = \"training_39-1\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Single-Task Execution Demo\n",
+    "Let's say we just kicked off a workflow like this, in which we wrote a **SageMaker SimpleTrainingJob task** to train **an XGBoost model** on a dataset.\n",
+    "![XGBoost workflow](wf-failed.png \"Failed workflow\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Oops! The TrainingJob task failed \n",
+    "\n",
+    "![Failed Execution](failed_exec.png \"Failed exec\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Let's take a look at the definition of the TrainingJob Task"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 55,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "alg_spec = training_job_models.AlgorithmSpecification(\n",
+    "    input_mode=_sdk_sagemaker_types.InputMode.FILE,\n",
+    "    algorithm_name=_sdk_sagemaker_types.AlgorithmName.XGBOOST,\n",
+    "    # ----->>>> BUGGY IMAGE VERSION !! <<<<------\n",
+    "    algorithm_version=\"0.72\",\n",
+    "    metric_definitions=[\n",
+    "        training_job_models.MetricDefinition(name=\"Minimize\", regex=\"validation:error\")\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "############################\n",
+    "# Define a TrainingJob task \n",
+    "############################\n",
+    "xgboost_train_task = training_job_task.SdkSimpleTrainingJobTask(\n",
+    "    training_job_config=training_job_models.TrainingJobConfig(\n",
+    "        instance_type=\"ml.m4.xlarge\",\n",
+    "        instance_count=1,\n",
+    "        volume_size_in_gb=25,\n",
+    "    ),\n",
+    "    algorithm_specification=alg_spec,\n",
+    "    cache_version='3',\n",
+    "    cacheable=True,\n",
+    ")\n",
+    "\n",
+    "# Defining inputs for the TrainingJob task\n",
+    "training_inputs={\n",
+    "    \"train\": \"s3://lyft-modelbuilder/test-datasets/pima/train\",\n",
```

Comment:
@EngHabu not a huge fan of having these out in the public. what do you think?

Reply:
agreed